bouncer/src/repo/manipulator/modelconvertor/import/odaHelper/file_processor_nwd.cpp

```diff
@@ -0,0 +1,821 @@
+			{
+				OdNwGeometryMeshPtr pGeometrMesh = OdNwGeometryMesh::cast(pFrag->getGeometryId().safeOpenObject());
+				if (!pGeometrMesh.isNull())
+				{
+					const OdArray<OdGePoint3d>& aVertices = pGeometrMesh->getVertexes();
+					const OdArray<OdGeVector3d>& aNormals = pGeometrMesh->getNormales();
+					const OdArray<OdGePoint2d>& aUvs = pGeometrMesh->getUVParameters();
+
+					const OdArray<OdNwTriangleIndexes>& aTriangles = pGeometrMesh->getTriangles();
+
+					if (pGeometrMesh->getIndexes().length() && !aTriangles.length())
+					{
+						repoError << "Mesh " << pNode->getDisplayName().c_str() << " has indices but does not have a triangulation. Only triangulated meshes are supported.";
+					}
+
+					for (auto triangle = aTriangles.begin(); triangle != aTriangles.end(); triangle++)
+					{			
+						std::vector<repo::lib::RepoVector3D64> vertices;
+						vertices.push_back(convertPoint(aVertices[triangle->pointIndex1], transformMatrix));
+						vertices.push_back(convertPoint(aVertices[triangle->pointIndex2], transformMatrix));
+						vertices.push_back(convertPoint(aVertices[triangle->pointIndex3], transformMatrix));
+
+						auto normal = calcNormal(vertices[0], vertices[1], vertices[2]);
+
+						std::vector<repo::lib::RepoVector2D> uvs;
+						if (aUvs.length()) 
+						{
+							uvs.push_back(convertPoint(aUvs[triangle->pointIndex1]));
+							uvs.push_back(convertPoint(aUvs[triangle->pointIndex2]));
+							uvs.push_back(convertPoint(aUvs[triangle->pointIndex3]));
+						}
+
+						context.collector->addFace(vertices, normal, uvs);
+					}					
+				}
+			}
+			else if (pFrag->isPointSet())
+			{
+				//Todo; warning unsupported geometry type
+			}
+			else if (pFrag->isText())
+			{
+				//Todo; warning unsupported geometry type
+			}
+			else if (pFrag->isTube())
+			{
+				//Todo; warning unsupported geometry type
+			}
+			else if (pFrag->isEllipse())
+			{
+				// Todo; warning unsupported geometry type
+			}
+		}
+	}
+
+	return eOk;
+}
+
+bool isInstanced(OdNwModelItemPtr pNode)
+{
+	if (pNode.isNull()) {
+		return false;
+	}
+	if (pNode->getIcon() == NwModelItemIcon::INSERT_GEOMETRY) {
+		return true;
+	}
+	if (pNode->getIcon() == NwModelItemIcon::INSERT_GROUP) {
+		return true;
+	}
+	return false;
+}
+
+bool isCollection(OdNwModelItemPtr pNode)
+{
+	if (pNode.isNull()){
+		return false;
+	}
+	if (pNode->getIcon() == NwModelItemIcon::COLLECTION) {
+		return true;
+	}
+	if (pNode->getIcon() == NwModelItemIcon::GROUP) {
+		return true;
+	}
+	return false;
+}
+
+// Traverses the scene graph recursively. Each invocation of this method operates 
+// on one OdNwModelItem, which is the basic OdNwObject that makes up the geometry 
+// scene/the tree in the Standard View.
+// As the scene is traversed we pass nodes that delimit branches, e.g. nodes that 
+// define the start of whole files. These stateful properties are stored in the 
+// context.
+
+OdResult traverseSceneGraph(OdNwModelItemPtr pNode, RepoNwTraversalContext context) 
```

Comment:
can we possibly split this file into 2 files?

So the other importers (Rvt, dgn etc) they all have a file_processor and a data_processor

file_processor is mostly just to init and setup everything (e.g. select the viewpoint, plug the file in, setup the devices etc

and data processor is the actual data extraction process.

It'd be good to be consistent so we know where to look for different situations!